### PR TITLE
#6Task.vladislavBirkos

### PR DESCRIPTION
**Pull request by dirtwitcher**
https://travis-ci.org/github/dirtwitcher/js-assignments/builds/711719934

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -82,10 +93,11 @@ function getSumBetweenNumbers(n1, n2) {
  *   10,10,10 =>  true
  */
 function isTriangle(a,b,c) {
-    throw new Error('Not implemented');
+    if (a**2 + b**2 === c**2 || a**2 + c**2 === b**2 || b**2 + c**2 === a**2 || (a === b && b === c)){
+        return true;
+    } return false;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: We can just `return (a**2 + b**2 === c**2 || a**2 + c**2 === b**2 || b**2 + c**2 === a**2 || (a === b && b === c))`

Please, see https://javascript.info/logical-operators

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -166,10 +189,15 @@ function isInsideCircle(circle, point) {
  *   'entente' => null
  */
 function findFirstSingleChar(str) {
-    throw new Error('Not implemented');
+    for (let i = 0; i < str.length; i++) {
+        let c = str.charAt(i);
+        if (str.indexOf(c) === i && str.indexOf(c, i + 1) === -1) {
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: we can improve this line `if (str.indexOf(c) === str.lastIndexOf(c)) {`

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -226,10 +268,9 @@ function reverseString(str) {
  *   34143 => 34143
  */
 function reverseInteger(num) {
-    throw new Error('Not implemented');
+    return reverseString(num);
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Using existed functions is ok in real projects, but here, the previous function might be changed for some reason.
At least reverseString returns string type. 
The number is expected as a result of reverseInteger (here it's converted automatically to number when compared to number in test)

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -296,10 +359,65 @@ function getDigitalRoot(num) {
...
+            if (str.charAt(i-1) === '<'){
+                br4--; 
+                str = str.slice(0,i-1) + str.slice(i+1);
+                i -= 2;
+                continue;
+            } else return false;
+        }
+        if (br1 < 0 || br2 < 0 || br3 < 0 || br4 < 0 ) return false;
+    }
+    return (br1 === 0 && br2 === 0 && br3 === 0 && br4 === 0 );
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, code style. (spaces)

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -356,10 +516,16 @@ function timespanToHumanString(startDate, endDate) {
...
 function toNaryString(num, n) {
-    throw new Error('Not implemented');
+    let abc="0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ".split('');
+    let s = '';
+	let mas = abc.slice(0,n);
+	while(num > 0){
+		s = String(s) + mas[num%n];
+		num = Math.floor(num/n);
+	}
+    return s.split('').reverse().join('');
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please, simplify solution using toString(n)

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -373,10 +539,23 @@ function toNaryString(num, n) {
...
+            if(pathes[i][j] === resultStr[j]){
+                bufResult += pathes[i][j];
+            } else {
+                break;
+            }
+        }
+        resultStr = bufResult;
+        bufResult = '';
+    }
+    return resultStr.slice(0,resultStr.lastIndexOf('/')+1);
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: code style

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by dirtwitcher** [comment]
https://travis-ci.org/github/dirtwitcher/js-assignments/builds/721632327

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**